crates/factor-key-value/src/spin_cli_resolver.rs

```diff
@@ -0,0 +1,47 @@
+use crate::runtime_config::RuntimeConfigResolver;
+use crate::store::{store_from_toml_fn, MakeKeyValueStore, StoreFromToml};
+use spin_key_value::StoreManager;
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Default)]
+pub struct SpinCliRuntimeConfigResolver {
```

Comment:
I think with a tweak (below) this could be a general purpose "delegating resolver" e.g. `DelegatingRuntimeConfigResolver`

Reply:
I agree that would be nice to make it generalized with that change too. Only other consideration is how to set the default DB directory. It may make sense to also set that in the resolver